src/daft-local-execution/src/sources/scan_task.rs

```diff
@@ -97,9 +111,80 @@ impl Source for ScanTaskSource {
     }
 }
 
+// Read all iceberg delete files and return a map of file paths to delete positions
+async fn get_delete_map(
+    scan_tasks: &[Arc<ScanTask>],
+) -> DaftResult<Option<HashMap<String, Vec<i64>>>> {
+    let delete_files = scan_tasks
+        .iter()
+        .flat_map(|st| {
+            st.sources
+                .iter()
+                .filter_map(|source| source.get_iceberg_delete_files())
+                .flatten()
+                .cloned()
+        })
+        .collect::<Vec<_>>();
```

Comment:
I think there may be delete files that are shared between scan tasks. Perhaps you should collect into a HashSet instead